proofs/eo/cpc/programs/Arith.eo

```diff
@@ -147,8 +153,8 @@
 ; - n Int: The number of times to multiply, expected to be a non-negative numeral.
 ; - a T: The term to muliply.
 ; return: The result of multiplying a, n times.
-(program $arith_unfold_pow_rec ((T Type) (n Int) (a T) (Any Type))
-  :signature (Int T) Any
+(program $arith_unfold_pow_rec ((T Type) (n Int) (a T))
+  :signature ((eo::quote n) T) (eo::ite (eo::eq n 0) Int T)
```

Comment:
The `ite` in the signature feels out of place. How about defining `$arith_mk_one ((T Type))` and using that in the definition instead?

Reply:
The issue is that the nil terminator for `*` is always `1` (integer), which is why this method (which returns intermediate terms in the construction of `a^n` has a somewhat strange type.